src/Services/QueuedJobService.php

```diff
@@ -731,7 +732,17 @@ public function runJob($jobId)
                     }
 
                     if (!$broken) {
-                        // Collect output as job messages as well as sending it to the screen
+                        // Inject real-time log handler
+                        $logger = Injector::inst()->get(LoggerInterface::class);
+                        if ($logger instanceof Logger) {
+                            $logger->pushHandler(new QueuedJobHandler($job, $jobDescriptor));
```

Comment:
```suggestion
                            $logger->pushHandler(QueuedJobHandler::create($job, $jobDescriptor));
```

Reply:
Do you think it might be worth checking whether the Logger already has this handler? I'm thinking about if you have a service running multiple jobs in the same process, it may end up stacking QueuedJobHandler instances into the logger, and aggregating old job information that isn't relevant after it's completed, e.g.:

- Run job 1

- Run job 1 (irrelevant now)
- Run job 2

- Run job 1 (irrelevant now)
- Run job 2 (irrelevant now)
- Run job 3

Reply:
Good point, I'll add this now.